### PR TITLE
Integrate local sbom extension

### PR DESCRIPTION
**Pull request by PeterSchafer**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
It integrates sbom as a local extension and improves the general application behaviour around extensions.

In addition some tests required adaption in the sense that they were expecting a certain order of messages being send which has now changed a bit. They have been either adapted to focus on certain messages or to avoid sending new messages.

#### Where should the reviewer start?
As usual start in cmd/cliv2/main.go

#### How should this be manually tested?
snyk sbom --experimental

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -159,6 +162,33 @@ func prepareRootCommand() *cobra.Command {
 	return &rootCommand
 }
 
+func doFallback(err error, helped bool) (fallback bool) {
+	fallback = false
+	preCondition := err != nil && helpProvided == false
+	if preCondition {
+		errString := err.Error()
```
Comment (Avishagp) [inline cliv2/cmd/cliv2/main.go:169]: Is it safe to check for the error string? isn't there an error code we can check against?

File: test/jest/acceptance/analytics.spec.ts [hunk 2]
```diff
@@ -48,12 +48,16 @@ describe('analytics module', () => {
 
     expect(code).toBe(0);
 
+    const requests = server.getRequests().filter((value) => {
```
Comment (Avishagp) [inline test/jest/acceptance/analytics.spec.ts:51]: How is this file's changes related?

File: cliv2/cmd/cliv2/main.go (hunk 1 above)
Comment (PeterSchafer) [reply cliv2/cmd/cliv2/main.go:169]: No it isn't. But I failed to find another way for now. I was struggling with this quite some time and decided to to for the string comparison for now and look for a better approach in the next increment. I will create tickets for the different tasks.

File: test/jest/acceptance/analytics.spec.ts (hunk 2 above)
Comment (PeterSchafer) [reply test/jest/acceptance/analytics.spec.ts:51]: This is one of the mentioned changes in tests that were required due to the changed number and order of requests being send.

File: cliv2/cmd/cliv2/main.go (hunk 1 above)
Comment (Avishagp) [reply cliv2/cmd/cliv2/main.go:169]: Great. Thanks!

**Review by Avishagp — APPROVED** [review]

**Review by mcombuechen — APPROVED** [review]

**Review by francescomari — APPROVED** [review]

**Review by Avishagp — APPROVED** [review]

**Pull request merged**